app/Http/Controllers/Appeal/PublicAppealController.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace App\Http\Controllers\Appeal;
+
+use App\Appeal;
+use App\Ban;
+use App\Http\Controllers\Controller;
+use App\Jobs\GetBlockDetailsJob;
+use App\Log;
+use App\MwApi\MwApiUrls;
+use App\Privatedata;
+use App\Rules\SecretEqualsRule;
+use App\Sendresponse;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log as LaravelLog;
+use Illuminate\Validation\Rule;
+use Symfony\Component\HttpFoundation\IpUtils;
+
+class PublicAppealController extends Controller
+{
+    public function store(Request $request)
+    {
+        $ua = $request->userAgent();
+        $ip = $request->ip();
+        $lang = $request->header('Accept-Language');
+
+        $data = $request->validate([
+            'appealtext' => 'required|max:4000',
+            'appealfor'  => 'required|max:100',
```

Comment:
```suggestion
            'appealfor'  => 'required|max:40',
```

Reply:
We don't need anything higher than that, most people over 40 chars don't appeal. We've been getting full block text inside this field which we shouldn't, so hope this prevents

Reply:
IPv6 addresses may technically be 45 chars... however those are v4 addresses encoded in v6 and MediaWiki does not support those addresses anyway. Might be necessary to include them but not sure.

https://stackoverflow.com/questions/166132/maximum-length-of-the-textual-representation-of-an-ipv6-address

Reply:
Also MediaWiki allows up to 85 chars. I've set it to 50, hopefully that is good enough.

https://github.com/wikimedia/operations-mediawiki-config/blob/66965597401b1d5bc4f9437c94457cfb51467374/wmf-config/CommonSettings.php#L343